src/app/shared/components/List/index.tsx

```diff
@@ -211,47 +211,47 @@ ListProps<T, F>): JSX.Element {
       className={`relative flex h-full flex-col overflow-x-hidden overflow-y-hidden ${className}`}
     >
       {/* HEAD */}
-      <div className="flex h-12 shrink-0 flex-row px-5">
-        {/* COLUMN */}
-        <div className="flex h-full min-w-full flex-row items-center border-b border-gray-10">
-          {/* SELECTION CHECKBOX */}
-          <div className="flex h-full flex-row items-center justify-between pr-4">
-            <BaseCheckbox
-              checked={selectedItems.length > 0}
-              indeterminate={items.length > selectedItems.length && selectedItems.length > 0}
-              onClick={onTopSelectionCheckboxClick}
-            />
-          </div>
-
-          {header.map((column) => (
-            <div
-              onClick={column.orderable ? () => onOrderableColumnClicked(column) : undefined}
-              key={column.name.toString()}
-              className={`flex h-full shrink-0  flex-row items-center space-x-1.5 text-base font-medium text-gray-60  ${
-                column.width
-              } ${column.orderable ? 'cursor-pointer hover:text-gray-80' : ''}`}
-            >
-              <span>{column.label}</span>
-              {column.name === orderBy?.field &&
-                column.orderable &&
-                (orderBy?.direction === 'ASC' ? (
-                  <ArrowUp size={14} weight="bold" />
-                ) : (
-                  <ArrowDown size={14} weight="bold" />
-                ))}
-            </div>
-          ))}
-          {isVerticalScrollbarVisible && <div className="mr-15px" />}
-          {menu && <div className="flex h-full w-12 shrink-0" />}
-        </div>
-      </div>
 
       {/* BODY */}
-      <div id="scrollableList" className="flex h-full flex-col overflow-x-auto overflow-y-auto" ref={ref}>
-        {(!hasMoreItems ?? false) && items.length === 0 && !isLoading ? (
-          emptyState
-        ) : items.length > 0 ? (
-          <>
+      {(!hasMoreItems ?? false) && items.length === 0 && !isLoading ? (
+        emptyState
+      ) : items.length > 0 ? (
+        <>
+          <div className="flex h-12 shrink-0 flex-row px-5">
+            {/* COLUMN */}
+            <div className="flex h-full min-w-full flex-row items-center border-b border-gray-10">
+              {/* SELECTION CHECKBOX */}
+              <div className="flex h-full flex-row items-center justify-between pr-4">
+                <BaseCheckbox
+                  checked={selectedItems.length > 0}
+                  indeterminate={items.length > selectedItems.length && selectedItems.length > 0}
+                  onClick={onTopSelectionCheckboxClick}
+                />
+              </div>
+
+              {header.map((column) => (
+                <div
```

Comment:
I think this is not the best way to hide the header in case you want to show only the empty state. 
It makes the code more cluttered and harder to distinguish between the header and the body of the list. 
If you want to use the same ternary to show the empty or not, I think it is a better alternative to remove the Header and Body to separate components (not necessarily in another file, they can be kept in this one) and put them together there, so each element remains separate and makes maintenance easier.

If you can think of another way that you consider good, that's fine by me.

Reply:
Maybe we can create a general view for all the views instead of using the same component for "Drive", "Recents", "Trash" and a different view for "Shared" since all the views use the List component but in different ways (Shared does use the empty prop but the others don't, for example). 

 I know this will take a lot of time because it is a big change in the code, but I think we can take advantage and use a kind of Wrapper for all the views we have and try to have it a little more centralized.

 If not, we can separate the header and body into different components, I agree with you.

Reply:
Drive, Recents and Trash are already using the same view component, the `DriveExplorer` one that uses this list.